structure.go

```diff
@@ -106,3 +106,42 @@ func (s Structure) UsePalette(name string) {
 	s.palette = &p
 	s.paletteName = name
 }
+
+// WithRotation returns a new structure with the same contents but rotated 90 degrees clockwise the specified
+// amount of times.
+func (s Structure) WithRotation(iterations int) Structure {
+	iterations = iterations % 4 // Make sure iterations is between 0 and 3.
+	sizeX, sizeY, sizeZ := int(s.Size[0]), int(s.Size[1]), int(s.Size[2])
+
+	var newStructure Structure
+	switch iterations {
+	case 0, 2:
```

Comment:
I would just return if there's no iterations.

Reply:
At 180 degrees it has rotations, I guess i'll remove the 0 degrees and return the same structure there instead